### PR TITLE
fix: add target_url and logs

### PR DESCRIPTION
**Pull request by apoorvsadana**

### CODE REVIEW

File: src/handlers/indexers/indexer_types/webhook.rs [hunk 1]
```diff
@@ -13,24 +14,43 @@ impl Indexer for WebhookIndexer {
...
 
         let id = child_handle.id();
 
         tokio::spawn(async move {
+            // all success messages are logged first, if error messages
+            // occur between success messages they won't be logged till
+            // the indexer is stopped
+            if let Some(ref mut stdout) = child_handle.stdout {
+                for line in BufReader::new(stdout).lines() {
+                    let line = line.unwrap();
```
Comment (EvolveArt) [inline src/handlers/indexers/indexer_types/webhook.rs:42]: unwrap safe ?

File: src/handlers/indexers/create_indexer.rs [hunk 2]
```diff
@@ -14,57 +15,77 @@ use crate::infra::repositories::indexer_repository::{self, IndexerDb};
...
+        let field_name = field.name().ok_or(IndexerError::InternalServerError)?;
+        match field_name {
+            "script.js" => {
+                create_indexer_request.data = field.bytes().await.map_err(IndexerError::FailedToReadMultipartField)?
+            }
+            "webhook_url" => {
+                create_indexer_request.webhook_url =
+                    field.text().await.map_err(IndexerError::FailedToReadMultipartField)?
+            }
+            &_ => return Err(IndexerError::UnexpectedMultipartField(field_name.to_string())),
```
Comment (EvolveArt) [inline src/handlers/indexers/create_indexer.rs:37]: is the `&` a typo?

File: src/handlers/indexers/create_indexer.rs [hunk 3]
```diff
@@ -14,57 +15,77 @@ use crate::infra::repositories::indexer_repository::{self, IndexerDb};
...
+        };
+    }
+    if create_indexer_request.data.is_empty() || create_indexer_request.webhook_url.is_empty() {
```
Comment (EvolveArt) [inline src/handlers/indexers/create_indexer.rs:40]: let's implement `is_empty` on the struct

File: src/handlers/indexers/start_indexer.rs [hunk 4]
```diff
@@ -62,6 +65,13 @@ pub async fn start_indexer(id: Uuid) -> Result<(), IndexerError> {
     Ok(())
 }
 
+pub async fn start_indexer_api(
```
Comment (EvolveArt) [inline src/handlers/indexers/start_indexer.rs:68]: we should check for the current indexer status

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: src/handlers/indexers/start_indexer.rs (hunk 4 above)
Comment (apoorvsadana) [reply src/handlers/indexers/start_indexer.rs:68]: this is already checked inside start_indexer

File: src/handlers/indexers/create_indexer.rs (hunk 3 above)
Comment (apoorvsadana) [reply src/handlers/indexers/create_indexer.rs]: done. I implemeted is_ready as I wanted to check all fields are available (whereas is_empty would check if all fields are empty)

File: src/handlers/indexers/create_indexer.rs (hunk 2 above)
Comment (apoorvsadana) [reply src/handlers/indexers/create_indexer.rs]: ya, fixed, but it still worked apparently

File: src/handlers/indexers/indexer_types/webhook.rs (hunk 1 above)
Comment (apoorvsadana) [reply src/handlers/indexers/indexer_types/webhook.rs]: handled in the new flow

File: src/handlers/indexers/start_indexer.rs (hunk 4 above)
Comment (EvolveArt) [reply src/handlers/indexers/start_indexer.rs:68]: right I mean we should add support for re-starting an indexer that failed running

Comment (apoorvsadana) [reply src/handlers/indexers/start_indexer.rs:69]: added!

**Pull request merged**